src/components/FavoriteCheckbox/FavoriteCheckbox.component.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Favorite from '@material-ui/icons/Favorite';
+import FavoriteBorder from '@material-ui/icons/FavoriteBorder';
+import { storage } from '../../utils/storage';
+
+const FavoriteCheckbox = function ({ id, snippet, etag, favorite }) {
+  const videoId = id.videoId;
+  const [fav, setFav] = useState(favorite);
+  const toggleFavorite = function (event) {
+    const favorites = storage.get('favorites') ?? {};
+
+    if (event.target.checked) {
```

Comment:
I recommend using destructuring and defaults or code in self defence because this line could raise errors if event is undefined for some reason, So,

```js
function ({ target = {} } = {})
```
then
```suggestion
    if (target.checked) {
```